agent/rpc/peering/service.go

```diff
@@ -227,57 +274,6 @@ func (s *Server) GenerateToken(
 		}
 	}
 
-	peeringOrNil, err := s.getExistingPeering(req.PeerName, entMeta.PartitionOrDefault())
-	if err != nil {
-		return nil, err
-	}
-
-	// validate that this peer name is not being used as a dialer already
-	if err = validatePeer(peeringOrNil, false); err != nil {
-		return nil, err
-	}
-
-	canRetry := true
-RETRY_ONCE:
-	id, err := s.getExistingOrCreateNewPeerID(req.PeerName, entMeta.PartitionOrDefault())
-	if err != nil {
-		return nil, err
-	}
-	writeReq := pbpeering.PeeringWriteRequest{
-		Peering: &pbpeering.Peering{
-			ID:   id,
-			Name: req.PeerName,
-			Meta: req.Meta,
-
-			// PartitionOrEmpty is used to avoid writing "default" in OSS.
-			Partition: entMeta.PartitionOrEmpty(),
-		},
-	}
-	if err := s.Backend.PeeringWrite(&writeReq); err != nil {
-		// There's a possible race where two servers call Generate Token at the
-		// same time with the same peer name for the first time. They both
-		// generate an ID and try to insert and only one wins. This detects the
-		// collision and forces the loser to discard its generated ID and use
-		// the one from the other server.
-		if canRetry && strings.Contains(err.Error(), "A peering already exists with the name") {
-			canRetry = false
-			goto RETRY_ONCE
-		}
-		return nil, fmt.Errorf("failed to write peering: %w", err)
-	}
-
-	q := state.Query{
-		Value:          strings.ToLower(req.PeerName),
-		EnterpriseMeta: *entMeta,
-	}
-	_, peering, err := s.Backend.Store().PeeringRead(nil, q)
-	if err != nil {
-		return nil, err
-	}
-	if peering == nil {
-		return nil, fmt.Errorf("peering was deleted while token generation request was in flight")
-	}
```

Comment:
I removed this check for the following reasons:
* Very unlikely scenario
* Peering may still be deleted after token generation

We can never eliminate this class of bug (it will eventually be caught by the dialer) so it's better to remove the false sense of security IMO. Let me know if there's reasons to keep it.